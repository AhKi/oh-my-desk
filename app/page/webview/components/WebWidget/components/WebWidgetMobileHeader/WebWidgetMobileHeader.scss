.WebWidgetMobileHeader {
  background-color: #7f7f7f;
  height: 26px;
  opacity: 0;
  position: fixed;
  transition: all .5s ease-in;
  width: 100%;

  &:hover {
    opacity: 0.8;
  }
}

.WebWidgetModalHeader__open {
  opacity: 0.8;
}

.WebWidgetMobileHeader__setting {
  color: #F1F1F1;
  float: right;
  height: 27px;
}

@media (min-width: 412px) {
  .WebWidgetMobileHeader {
    display: none;
  }
}
