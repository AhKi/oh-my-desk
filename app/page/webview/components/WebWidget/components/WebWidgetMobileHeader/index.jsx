import React from 'react';
import PropTypes from 'prop-types';
import HistoryGoBackButton from '../Button/HistoryGoBackButton';
import ReloadButton from '../Button/ReloadButton';
import './WebWidgetMobileHeader.scss';

const propTypes = {
  isLoading: PropTypes.bool,
  webView: PropTypes.object, // eslint-disable-line react/forbid-prop-types
  onGoBack: PropTypes.func,
  onGoForward: PropTypes.func,
  onRefresh: PropTypes.func,
  onStopRefresh: PropTypes.func,
};
const defaultProps = {
  isLoading: false,
  webView: null,
  onGoBack() {},
  onGoForward() {},
  onRefresh() {},
  onStopRefresh() {},
};

class WebWidgetMobileHeader extends React.Component {
  render() {
    const {
      webView,
      isLoading,
      onGoBack,
      onGoForward,
      onRefresh,
      onStopRefresh,
    } = this.props;

    return (
      <div
        className="WebWidgetMobileHeader"
      >
        <HistoryGoBackButton
          isCanGoBack={webView && webView.canGoBack()}
          isCanGoForward={webView && webView.canGoForward()}
          onGoBack={onGoBack}
          onGoForward={onGoForward}
        />
        <ReloadButton
          isLoading={isLoading}
          onRefresh={onRefresh}
          onStopRefresh={onStopRefresh}
        />
      </div>
    );
  }
}

WebWidgetMobileHeader.propTypes = propTypes;
WebWidgetMobileHeader.defaultProps = defaultProps;

export default WebWidgetMobileHeader;
