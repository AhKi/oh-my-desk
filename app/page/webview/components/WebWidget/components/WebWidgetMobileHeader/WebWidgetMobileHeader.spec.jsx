import React from 'react';
import { shallow } from 'enzyme';

import WebWidgetMobileHeader from './';

describe('<WebWidgetMobileHeader />', () => {
  it('should match to snapshot when render default', () => {
    const webView = {
      canGoBack() {},
      canGoForward() {},
    };
    const wrapper = shallow(
      <WebWidgetMobileHeader
        webView={webView}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
