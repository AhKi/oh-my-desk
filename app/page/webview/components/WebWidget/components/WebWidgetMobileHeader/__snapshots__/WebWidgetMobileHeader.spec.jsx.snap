// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WebWidgetMobileHeader /> should match to snapshot when render default 1`] = `
<div
  className="WebWidgetMobileHeader"
>
  <HistoryGoBackButton
    className=""
    isCanGoBack={false}
    isCanGoForward={false}
    onGoBack={[Function]}
    onGoForward={[Function]}
  />
  <ReloadButton
    className="ReloadButton"
    isLoading={false}
    onRefresh={[Function]}
    onStopRefresh={[Function]}
  />
</div>
`;
